@shared/api/internal.ts

```diff
@@ -545,6 +566,48 @@ export const submitFreighterTransaction = async ({
   return await server.submitTransaction(tx);
 };
 
+export const submitFreighterSorobanTransaction = async ({
+  signedXDR,
+  networkDetails,
+}: {
+  signedXDR: string;
+  networkDetails: NetworkDetails;
+}) => {
+  let tx = {} as SorobanClient.Transaction | SorobanClient.FeeBumpTransaction;
+
+  try {
+    tx = SorobanClient.TransactionBuilder.fromXDR(
+      signedXDR,
+      networkDetails.networkPassphrase,
+    );
+  } catch (e) {
+    console.error(e);
+  }
+
+  const server = new SorobanClient.Server(SOROBAN_RPC_URLS.futureNet, {
+    allowHttp: true,
+  });
+
+  // TODO: fixed in Sorobanclient, not yet released
+  let response = (await server.sendTransaction(tx)) as any;
+
+  try {
+    // Poll this until the status is not "pending"
+    while (response.status === "pending") {
```

Comment:
`pending` should probably be part of an enum. That's generally the pattern we use for dealing with statuses

Reply:
Added the enum in d172235d8ea857ddb35022b80367ea7dbba8ff71